tastyworks/models/watchlists.py

```diff
@@ -0,0 +1,95 @@
+import aiohttp
+from tastyworks.models.session import TastyAPISession
+
+
+class Watchlist(object):
+    def __init__(self, name=None):
+        self.name = name
+        self.group_name = None
+        self.securities = {}
+
+    @classmethod
+    def from_list(cls, data: list):
+        """From List
+
+        Generator for making a watchlist.
+
+        Notes:
+            There is an instability issue with Tastyworks API.  Hence
+            the need to catch an exception.  Special symbols don't have
+            instrument-type's at all.
+
+        Args:
+            data (list):   List of dict objects containing symbol &
+                           [instrument-type | instrument_type] keys.
+
+        Returns:
+            Watchlist:  A watchlist object
+        """
+        inst = cls()
+        for item in data:
+            try:
+                inst.securities[item['symbol']] = {
+                    'instrument-type': item['instrument-type']
+                }
+            except KeyError:
+                try:
+                    inst.securities[item['symbol']] = {
+                        'instrument_type': item['instrument_type']
+                    }
+                except KeyError:
+                    inst.securities[item['symbol']] = {
+                        'instrument-type': None
+                    }
+        return inst
+
+
+class WatchlistGroup(object):
+    """Watchlist Group
+
+    Generator for making document of watchlist objects.
+
+    Notes:
+        Not all watchlists return group-name key's.  Hence the need to catch
+        an exception.
+
+    Args:
+        session (TastyAPISession): Tastyworks API Session object
+        public (boolean): True if retrieving public-watchlists, false if
+                          retrieving custom-watchlists
+
+    Returns:
+        WatchlistGroup: A group of watchlists
+
+    """
+    def __init__(self):
+        self.watchlists = {}
+
+    async def load_watchlists(
+        self, session: TastyAPISession, public: bool = True
+    ):
+        url = f'{session.API_url}/public-watchlists' if public else f'{session.API_url}/watchlists'  # NOQA: E501
+
+        async with aiohttp.request(
+            'GET', url, headers=session.get_request_headers()
+        ) as resp:
+            if resp.status != 200:
+                raise Exception('Could not get public asset watchlists')
+            data = await resp.json()
+
+        data = data['data']['items']
+        for entry in data:
+            list_data = entry['watchlist-entries']
+            wlist = Watchlist.from_list(list_data)
+            wlist.name = entry['name']
+            try:
+                wlist.group_name = entry['group-name']
+            except KeyError:
+                pass
+            self.watchlists[wlist.name] = wlist
+
+        return self
```

Comment:
If I understand correctly, this is returning and instance of `WatchlistGroup`? If so, should we not make this method a class method and then it can instantiate a new `WatchlistGroup` and return that?

Reply:
This is actually a bit over my head.  This was a result of me limiting my scope on this change.

This is the first time I've ever seen an object instantiated this way and it was a bit perplexing to me.  I'll read up some more on this subject today and get it updated.

Reply:
Check line 11 above :) The idea being that a classmethod belongs to the class (i.e. the prototype) rather than each instance (normal methods with `self`). It can therefore be used like a 'factory' (there's a design pattern denoting this approach) which gives you a new instance of the class every time you call it.

example usage if `load_watchilists` (I'd rename it to something like `get_watchlists` if converting to classmethod):
```
my_watchlists = WatchlistGroup.get_watchlists(session, public)
```

where `my_watchlists` is now an instaces of the `WatchlistGroup` class

Reply:
1. You blew my mind with this.

I honestly don't know how I didn't connect these dots before.  I've dealt with factories in the past but not at this level.  I ended up going down such a good rabbit trail for like 2 hours, that I ended up learning about descriptors and found quite the use for them here given the way watchlists are handled within their system.